### PR TITLE
Merge YelpReviewPolarity and YelpReviewFull Mocked Unit Tests

### PR DESCRIPTION
**Pull request by VirgileHlav**
Reference Issue: #1493

## Summary
- Merge mocked unit test for YelpReviewPolarity and YelpReviewFull dataset 

## Test
- ` pytest test/datasets/test_yelpreviews.py`

## Left to do
- revert #1568

### CODE REVIEW

File: test/datasets/test_yelpreviewpolarity.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+                dataset_line = (label, f"{rand_string}")
+                f.write(f'"{label}","{rand_string}"\n')
+
+                # append line to correct dataset split
+                mocked_lines.append(dataset_line)
+                seed += 1
+
+    compressed_dataset_path = os.path.join(base_dir, "yelp_review_polarity_csv.tar.gz")
+    # create gz file from dataset folder
+    with tarfile.open(compressed_dataset_path, "w:gz") as tar:
```
Comment (Nayef211) [inline test/datasets/test_yelpreviewpolarity.py:43]: Same comment as https://github.com/pytorch/text/pull/1566/files#r798933661

File: test/datasets/test_yelpreviewpolarity.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    os.makedirs(temp_dataset_dir, exist_ok=True)
+
+    seed = 1
+    mocked_data = defaultdict(list)
+    for file_name in ["train.csv", "test.csv"]:
+        csv_file = os.path.join(temp_dataset_dir, file_name)
+        mocked_lines = mocked_data[os.path.splitext(file_name)[0]]
+        with open(csv_file, "w") as f:
+            for i in range(5):
+                label = seed % 11
```
Comment (Nayef211) [inline test/datasets/test_yelpreviewpolarity.py:30]: YelpReviewPolarity has labels ranging from 1-2. This should be `label = seed % 2 + 1`

File: test/datasets/test_yelpreviewpolarity.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+from ..common.case_utils import TempDirMixin, zip_equal
+from ..common.torchtext_test_case import TorchtextTestCase
+
+
+def _get_mock_dataset(root_dir):
+    """
+    root_dir: directory to the mocked dataset
+    """
+    base_dir = os.path.join(root_dir, "YelpReviewPolarity")
+    temp_dataset_dir = os.path.join(base_dir, "yelp_review_polarity_csv")
```
Comment (Nayef211) [inline test/datasets/test_yelpreviewpolarity.py:20]: Same comment as https://github.com/pytorch/text/pull/1566/files#r798932093

**Comment by parmeet** [comment]
Overall LGTM, but I think we might be able to combine YelpReviewPolarity with YelpReviewFull since they are almost similar except for number of classes and compressed filename. We could potentially pass these as arguments to `_get_mock_dataset` function. We could potentially combine AmazonReviewFull and AmazonReviewPolarity and I think @Nayef211 is going to combine Squad1 and Squad2 as well.

**Comment by Nayef211** [comment]
> Overall LGTM, but I think we might be able to combine YelpReviewPolarity with YelpReviewFull since they are almost similar except for number of classes and compressed filename. We could potentially pass these as arguments to `_get_mock_dataset` function. We could potentially combine AmazonReviewFull and AmazonReviewPolarity and I think @Nayef211 is going to combine Squad1 and Squad2 as well.

@VirgileHlav you can look at https://github.com/pytorch/text/pull/1575 for an example of how we parameterized the squad datasets

**Review by parmeet — APPROVED** [review]
LGTM! Thanks for merging the tests.

**Pull request merged**